packages/hash/api/src/model/knowledge/linkEntity.model.ts

```diff
@@ -166,21 +167,65 @@ export default class extends EntityModel {
   }
 
   /**
-   * Update the link without modifying the indexes of its sibling links.
+   * Update an entity.
+   *
+   * @param params.properties - the properties object of the entity
+   * @param params.actorId - the id of the account that is updating the entity
+   */
+  async update(
+    graphApi: GraphApi,
+    params: {
+      properties: PropertyObject;
+      linkOrder: EntityLinkOrder;
+      actorId: string;
+    },
+  ): Promise<EntityModel> {
+    const { properties, actorId, linkOrder } = params;
+    const { baseId, entityTypeModel } = this;
+
+    const { data: metadata } = await graphApi.updateEntity({
+      actorId,
+      entityId: baseId,
+      entityTypeId: entityTypeModel.schema.$id,
+      properties,
+      leftOrder: linkOrder.leftOrder,
+      rightOrder: linkOrder.rightOrder,
+    });
+
+    return LinkEntityModel.fromEntity(graphApi, {
+      metadata: metadata as EntityMetadata,
+      properties,
+    });
+  }
+
+  /**
+   * Update the link without modifying the indices of its sibling links.
    *
    * @todo: deprecate this method when the Graph API handles updating the sibling indexes
    * @see https://app.asana.com/0/1200211978612931/1203031430417465/f
    *
-   * @param params.updatedIndex - the updated index of the link
+   * @param params.linkOrder {EntityLinkOrder} - the new orders to update for the link
    * @param params.actorId - the id of the account that is updating the link
    */
-  private async updateLink(
-    _graphApi: GraphApi,
-    _params: { updatedLeftOrder?: number; updatedRightOrder?: number },
+  async updateOrder(
+    graphApi: GraphApi,
+    params: { linkOrder: EntityLinkOrder; actorId: string },
   ) {
-    /**
-     * @todo: implement this when `updateEntity` can update the link metadata of a link
-     * @see https://app.asana.com/0/1202805690238892/1203384069111429/f
-     */
+    const { actorId, linkOrder } = params;
+    const { baseId, entityTypeModel, properties } = this;
+
+    const { data: metadata } = await graphApi.updateEntity({
+      actorId,
+      entityId: baseId,
+      entityTypeId: entityTypeModel.schema.$id,
+      properties,
+      leftOrder: linkOrder.leftOrder,
+      rightOrder: linkOrder.rightOrder,
+    });
```

Comment:
Here we risk overwriting the `properties` accidentally if the `properties` stored on the model class are outdated. Not sure how to fix this now, but perhaps worth capturing in a warning comment or `@todo` here if you can come up with one.

Reply:
I've captured it in this task https://app.asana.com/0/1201095311341924/1202890446280565/f